src/js/components/DataTable/v2/DataTableHeader.jsx

```diff
@@ -3,16 +3,19 @@ import React from 'react';
 import { flexRender } from '@tanstack/react-table';
 import PropTypes from 'prop-types';
 
-const DataTableHeader = ({ headerGroups }) => (
+const DataTableHeader = ({ headerGroups, columns }) => (
   <div className="rt-thead -header">
     <div className="rt-tr">
       {headerGroups
         .map((headerGroup) => (
-          headerGroup.headers.map((header) => (
-            <div className="w-100" key={header.id}>
-              {flexRender(header.column.columnDef.header, header.getContext())}
-            </div>
-          ))
+          headerGroup.headers.map((header, index) => {
+            const className = header.column.columnDef.meta?.getCellContext?.().className;
```

Comment:
The flexWidth can be passed as a meta in the same way the className is passed. So you don't have to pass columns as a prop.